std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/LongStorage.java

```diff
@@ -198,4 +187,70 @@ public LongStorage widen(IntegerType widerType) {
     assert widerType.fits(getType());
     return new LongStorage(data, (int) getSize(), getIsNothingMap(), widerType);
   }
+
+  /** Allow access to the underlying data array for copying. */
+  public long[] getArray() {
+    return data;
+  }
+
+  @Override
+  public ColumnLongStorageIterator iterator() {
+    return new LongStorageIterator(data, isNothing, (int) getSize());
+  }
+
+  private static class LongStorageIterator implements ColumnLongStorageIterator {
+    private final long[] data;
+    private final BitSet isNothing;
+    private final int size;
+    private int index = -1;
+
+    public LongStorageIterator(long[] data, BitSet isNothing, int size) {
+      this.data = data;
+      this.isNothing = isNothing;
+      this.size = size;
+    }
+
+    @Override
+    public Long getItemBoxed() {
+      return isNothing.get(index) ? null : data[index];
+    }
+
+    @Override
+    public long getItemAsLong() {
```

Comment:
Why do we continue to have this "boxed" vs. "unboxed" duality? Boxing doesn't influence the _peak performance_!